softwareconfiguration/mlplan/weka/src/example/java/ai/libs/mlplan/examples/multiclass/weka/MLPlanGraphVisualizationExample.java

```diff
@@ -1,63 +1,93 @@
+				.withNodeEvaluationTimeOut(new Timeout(30, TimeUnit.SECONDS)).withDataset(split.get(0));
+		MLPlan<IWekaClassifier> mlplan = mlplanBuilder.build();
+
+		/* create visualization */
+		AlgorithmVisualizationWindow window = new AlgorithmVisualizationWindow(mlplan);
+		window.withMainPlugin(new GraphViewPlugin());
+		window.withPlugin(new NodeInfoGUIPlugin(new JaicoreNodeInfoGenerator<>(new TFDNodeInfoGenerator())), new SearchRolloutHistogramPlugin(), new NodeInfoGUIPlugin(new TFDNodeAsCIViewInfoGenerator(mlplanBuilder.getComponents())),
+				new SolutionPerformanceTimelinePlugin(new HASCOSolutionCandidateRepresenter()), new HASCOModelStatisticsPlugin());
+
+		try {
+			long start = System.currentTimeMillis();
+			IWekaClassifier optimizedClassifier = mlplan.call();
+			long trainTime = (int) (System.currentTimeMillis() - start) / 1000;
+			System.out.println("Finished build of the classifier. Training time was " + trainTime + "s.");
+
+			/* evaluate solution produced by mlplan */
+			SupervisedLearnerExecutor executor = new SupervisedLearnerExecutor();
+			ILearnerRunReport report = executor.execute(optimizedClassifier, split.get(1));
+			System.out.println("Error Rate of the solution produced by ML-Plan: " + EClassificationPerformanceMeasure.ERRORRATE.loss(report.getPredictionDiffList()));
+		} catch (NoSuchElementException e) {
+			System.out.println("Building the classifier failed: " + e.getMessage());
+		}
+
+		/* obtain the algorithm history from the window and create a serializer */
+		AlgorithmEventHistory history = window.getAlgorithmEventHistory();
+
+		AlgorithmEventHistorySerializer serializer = new AlgorithmEventHistorySerializer();
+
+		/* serialize the obtained algorithm history and write it to a file */
+		String serializedHistory = serializer.serializeAlgorithmEventHistory(history);
+		FileUtil.writeFileAsList(Arrays.asList(serializedHistory), "history.json");
+
+		// read a serzialized history and start an algorithm inspector on this history
+
+		// AlgorithmEventHistorySerializer serializer = new AlgorithmEventHistorySerializer();
+		// AlgorithmEventHistory history = serializer.deserializeAlgorithmEventHistory(new File("data/keks.txt"));
+		//
+		// MLPlanWekaBuilder mlplanBuilder = new MLPlanWekaBuilder().withNumCpus(2).withTimeOut(new Timeout(120, TimeUnit.SECONDS)).withCandidateEvaluationTimeOut(new Timeout(30, TimeUnit.SECONDS))
+		// .withNodeEvaluationTimeOut(new Timeout(900, TimeUnit.SECONDS));
+		//
+		// AlgorithmVisualizationWindow window = new AlgorithmVisualizationWindow(history);
+		// window.withMainPlugin(new GraphViewPlugin());
+		// window.withPlugin(new NodeInfoGUIPlugin(new JaicoreNodeInfoGenerator<>(new TFDNodeInfoGenerator())), new SearchRolloutHistogramPlugin(), new NodeInfoGUIPlugin(new TFDNodeAsCIViewInfoGenerator(mlplanBuilder.getComponents())),
+		// new SolutionPerformanceTimelinePlugin(new HASCOSolutionCandidateRepresenter()));
```

Comment:
This should be resolved.

Reply:
It causes problems when you have two windows open at the same time. Hence, the second part should only be run when the first window is closed.